pkg/agent/memberlist/cluster_test.go

```diff
@@ -0,0 +1,679 @@
+//cluster_test.go:612: Node: node2, egressNum: 32
+//cluster_test.go:612: Node: node3, egressNum: 38
+//=== RUN   TestCluster_ShouldSelectByConsistentHash/select_node_from_alive_nodes-1000-egresses
+//=== CONT  TestCluster_ShouldSelectByConsistentHash
+//cluster_test.go:612: Node: node1, egressNum: 345
+//cluster_test.go:612: Node: node2, egressNum: 297
+//cluster_test.go:612: Node: node3, egressNum: 358
+//=== RUN   TestCluster_ShouldSelectByConsistentHash/select_node_from_alive_nodes-10000-egresses
+//=== CONT  TestCluster_ShouldSelectByConsistentHash
+//cluster_test.go:612: Node: node1, egressNum: 3437
+//cluster_test.go:612: Node: node2, egressNum: 2930
+//cluster_test.go:612: Node: node3, egressNum: 3633
+//--- PASS: TestCluster_ShouldSelectByConsistentHash (0.10s)
+//https://github.com/golang/groupcache/issues/29
+func TestCluster_ShouldSelectEgress(t *testing.T) {
+	nodes := []string{"node1", "node2", "node3"}
+	consistentHash := consistenthash.New(defaultVirtualNodeReplicas, nil)
+	consistentHash.Add(nodes...)
+
+	nodeSelectedForEgress := func(consistentHash *consistenthash.Map, egressName, nodeName string) bool {
+		return consistentHash.Get(egressName) == nodeName
+	}
+
+	checkNum := func(count int, myNode string) int {
+		totalNum := 0
+		for i := 0; i < count; i++ {
+			egressName := fmt.Sprintf("egress-%d", i)
+			if nodeSelectedForEgress(consistentHash, egressName, myNode) {
+				totalNum++
+			}
+		}
+		return totalNum
+	}
+
+	checkSum := func(egressNum int) int {
+		count := 0
+		for _, node := range nodes {
+			num := checkNum(egressNum, node)
+			count += num
+			t.Logf("Node: %s, egressNum: %d", node, num)
+		}
+		return count
+	}
+
+	testCases := []struct {
+		name      string
+		egressNum int
+	}{
+		{
+			name:      "select node from alive nodes",
+			egressNum: 3,
+		},
+		{
+			name:      "select node from alive nodes",
+			egressNum: 10,
+		},
+		{
+			name:      "select node from alive nodes",
+			egressNum: 100,
+		},
+		{
+			name:      "select node from alive nodes",
+			egressNum: 1000,
+		},
+		{
+			name:      "select node from alive nodes",
+			egressNum: 10000,
+		},
+	}
+	for _, tCase := range testCases {
+		t.Run(fmt.Sprintf("%s-%d-egresses", tCase.name, tCase.egressNum), func(t *testing.T) {
+			assert.Equal(t, tCase.egressNum, checkSum(tCase.egressNum))
+		})
+	}
+}
+
+//BenchmarkCluster_ShouldSelect
+//BenchmarkCluster_ShouldSelect/select_node_from_1000_alive_nodes
+//BenchmarkCluster_ShouldSelect/select_node_from_1000_alive_nodes-16         	    4860	    244613 ns/op
+//BenchmarkCluster_ShouldSelect/select_node_from_100_alive_nodes
+//BenchmarkCluster_ShouldSelect/select_node_from_100_alive_nodes-16          	   52707	     22412 ns/op
+//BenchmarkCluster_ShouldSelect/select_node_from_10_alive_nodes
+//BenchmarkCluster_ShouldSelect/select_node_from_10_alive_nodes-16           	  538476	      2273 ns/op
+//PASS
+//BenchmarkCluster_ShouldSelectHitNodeByConsistentHash
```

Comment:
what's BenchmarkCluster_ShouldSelectHitNodeByConsistentHash? Is it actually `BenchmarkCluster_ShouldSelect`  below? To avoid confusion, can you remove non-existing methods/tests and just comment consistentHash is more efficient than the way that hashs and sorts all items to get the first item?

Reply:
done